### ISSUE TITLE
See gloot internals when saving branch as scene, or making local.

### ISSUE THREAD

**Issue by kyle-wannacott** [peter-kish/gloot#153]
Hi,

Just curious, when saving a scene as a branch containing nodes from gloot, or "make local" a branch scene containing nodes from gloot; you can see the internal nodes. I don't know if it's supposed to be that way? or they are suppose to be hidden away from the user (like they are initially)?

Not really an issue per say... guess it could get cluttered if you have a ton of inventories

Cheers.

![image](https://github.com/peter-kish/gloot/assets/49783296/517fb1a8-fe52-4f2c-a1af-62d30ede5616)

**Comment by peter-kish** [comment]
Yeah, this is unintentional and I'm not sure what I can do about it...

The UI controls of the plugin are naturally composed out of built-in controls and those usually don't show up in the editor scene tree, until you save the scene as a branch - so I guess Godot is somewhat inconsistent on how it treats them.
I could simply not create the child nodes when inside the editor (`Engine.is_editor_hint()`), but then the controls will not be rendered inside the editor at all, which is even worse.

Thanks for letting me know, I wasn't aware of this issue at all. I'll see if I can come up with some fix for this...

**Comment by peter-kish** [comment]
Good news: It's a known Godot issue that has been fixed in the meantime: https://github.com/godotengine/godot/issues/82756
Bad news: Doesn't seem like it's not going to be released until Godot v4.3

**Comment by LeeWannacott** [comment]
Not a big deal I guess, At least there is a work around just don't save as a branch, or make local :sweat_smile: 

Always waiting for godot :robot:
